From 33bd697a44a5babf273ac2ed5d3e2891a95b516a Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Fri, 28 Sep 2018 15:33:34 +0800
Subject: [PATCH 4758/5242] MLK-15975-3: ASoC: fsl_sai: The offset of fifo_off
 is changed

commit  0b8df30dce42d3dbfcb17ed5e5fae6f959038ab2 from
https://source.codeaurora.org/external/imx/linux-imx.git

Commit 786c8bd56324 ("MLK-19734-3: dmaengine: imx-sdma: change
fifo offset of fifo_num") change the offset of fifo_off, so
the sai driver need to be updated.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
(cherry picked from commit c94ce8776e01f1f40a866d4da89603ab042dde0f)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_sai.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sound/soc/fsl/fsl_sai.c b/sound/soc/fsl/fsl_sai.c
index 3b4236a..c464cbc8 100644
--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -724,19 +724,19 @@ static int fsl_sai_hw_params(struct snd_pcm_substream *substream,
 						FSL_SAI_MAXBURST_TX * pins;
 				if (sai->is_dsd)
 					sai->dma_params_tx.fifo_num = pins +
-					   (sai->dataline_off_dsd[tx] << 8);
+					   (sai->dataline_off_dsd[tx] << 4);
 				else
 					sai->dma_params_tx.fifo_num = pins +
-						(sai->dataline_off[tx] << 8);
+						(sai->dataline_off[tx] << 4);
 			} else {
 				sai->dma_params_rx.maxburst =
 					FSL_SAI_MAXBURST_RX * pins;
 				if (sai->is_dsd)
 					sai->dma_params_rx.fifo_num = pins +
-					  (sai->dataline_off_dsd[tx] << 8);
+					  (sai->dataline_off_dsd[tx] << 4);
 				else
 					sai->dma_params_rx.fifo_num = pins +
-						(sai->dataline_off[tx] << 8);
+						(sai->dataline_off[tx] << 4);
 			}
 		}
 
-- 
1.7.9.5

